src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -27,41 +36,74 @@ const noChildError = new Error(LTIError.NoChild);
 })
 export class LTIComponent implements OnDestroy {
 
+  private loginId$ = this.activatedRoute.queryParamMap.pipe(map(queryParams => queryParams.get(loginIdParam)));
+
+  private isRedirection$ = this.activatedRoute.queryParamMap.pipe(
+    map(queryParams => queryParams.has(isRedirectionParam)),
+  );
+
   private contentId$ = this.activatedRoute.paramMap.pipe(
     map(params => {
       const contentId = params.get('contentId');
       if (!contentId) throw new Error('unexpected: contentId should be defined');
       return contentId;
     }),
   );
-  readonly navigationData$ = this.contentId$.pipe(
-    switchMap(contentId => this.getNavigationData(contentId).pipe(mapToFetchState())),
+
+  private isLoggedIn$ = this.loginId$.pipe(
+    switchMap(loginId => (loginId ? this.checkLoginService.check(loginId) : of(false))),
```

Comment:
Do you support the case where the loginId is not given? (in another way than just sending the user to the login-module.. which should not fix the problem.

Reply:
Fixed by displaying a login error. Is it ok ?
I figured that opening the app as LTI provider but no with no loginId does not make sense, but I'm not entirely sure about that.

Reply:
I also added test case 4 for that